test/openssl/test_pkey.rb

```diff
@@ -151,4 +151,27 @@ def test_x25519
     assert_equal bob_pem, bob.public_to_pem
     assert_equal [shared_secret].pack("H*"), alice.derive(bob)
   end
+
+  def test_compare?
+    same_keys = {rsa: "rsa1024", dh: "dh1024", dsa: "dsa256", ec: "p256"}
+    different_keys = {rsa: "rsa2048", dh: "dh-1", dsa: "dsa512", ec: "secp224r1"}
```

Comment:
I think testing RSA only is enough.

Reply:
Will remove, didn't think the extras would hurt.